libminifi/test/archive-tests/MergeFileTests.cpp

```diff
@@ -111,8 +111,8 @@ std::vector<FixedBuffer> read_archives(const FixedBuffer& input) {
   class ArchiveEntryReader {
    public:
     explicit ArchiveEntryReader(archive* arch) : arch(arch) {}
-    int read(uint8_t* out, std::size_t len) {
-      return gsl::narrow<int>(archive_read_data(arch, out, len));
+    size_t read(uint8_t* out, std::size_t len) {
+      return gsl::narrow_cast<size_t>(archive_read_data(arch, out, len));
```

Comment:
I know this is just a test file helper, but it would be safer to explicitly test for -1 and then use `gsl::narrow`, instead of using `gsl::narrow_cast` and risk that the return value will be something like `static_cast<size_t>(-3)`, which is not handled by `FixedBuffer::write()`.

Reply:
Introduced a check for negative return values in 0c23d4e